lib/board.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Board do
+
+  defstruct [:size, :cells, :marks]
+
+  def create_cells(size) do
+    Enum.to_list(1..size * size)
+  end
+
+  def place_mark(board, position, mark) do
+    %{board | cells: List.replace_at(board.cells, position - 1, mark)}
+  end
+
+  def rows(board) do
+    Enum.chunk_every(board.cells, board.size)
+  end
+
+  def columns(board) do
+    rows(board)
+    |> List.zip
+    |> Enum.map(&Tuple.to_list/1)
+  end
+
+  def diagonal_lines(board) do
+    [slice_diagonally(board.cells, board.size),
+     slice_diagonally(List.flatten(reverse_lines(rows(board))), board.size)]
+  end
+
+  defp all_lines(board) do
+    rows(board) ++ columns(board) ++ diagonal_lines(board)
+  end
+
+  def winning?(board) do
+    Enum.any?(all_lines(board), fn(line) -> winning_line?(line) end )
+  end
+
+  def winner(board) do
+    Enum.filter(all_lines(board), fn(line) -> winning_line?(line) end)
```

Comment:
Same here, I would go all-in with the `|>` syntax:
```
board
|> all_lines()
|> Enum.filter(winning_line?)
|> List.flatten
|> List.first
```
I am not even 100% you need the flatten though...

Reply:
Yeah, you need the flatten because `Enum.filter` returns this `[[x, x, x]]`

Reply:
or twice `Enum.first`